File: types/order.go
package types

import (
	"time"
)

type OrderStatus string

const (
	OrderPending   OrderStatus = "pending"
	OrderPaid      OrderStatus = "paid"
	OrderFulfilled OrderStatus = "fulfilled"
	OrderShipped   OrderStatus = "shipped"
	OrderDelivered OrderStatus = "delivered"
	OrderRefunded  OrderStatus = "refunded"
	OrderCanceled  OrderStatus = "canceled"
)

type Order struct {
	ID             string      `json:"id"`
	UserID         string      `json:"-"`
	Address        Address     `json:"address"`
	Amount         int64       `json:"amount"`
	TaxAmount      int64       `json:"tax_amount"`
	ShippingAmount int64       `json:"shipping_amount"`
	TotalAmount    int64       `json:"total_amount"`
	Status         OrderStatus `json:"status"`
	Items          []OrderItem `json:"items"`
	CreatedAt      time.Time   `json:"created_at"`
	UpdatedAt      time.Time   `json:"updated_at"`
}

type OrderItem struct {
	Product   Product `json:"product"`
	Thumbnail string  `json:"thumbnail"`
	AltText   string  `json:"alt_text"`
	Quantity  int     `json:"quantity"`
	UnitPrice int64   `json:"unit_price"`
}

File: web/src/types/order.ts
import type { Address } from './address'
import type { Product } from './product'

export interface OrderItem {
  product: Product
  thumbnail: string
  alt_text: string
  quantity: number
  unit_price: number
}

export type OrderStatus = 'pending' | 'paid' | 'refunded' | 'shipped' | 'delivered' | 'canceled'

export interface Order {
  id: string
  user_id: string
  address: Address
  items: OrderItem[]
  status: OrderStatus
  amount: number
  tax_amount: number
  shipping_amount: number
  total_amount: number
  created_at: string
  updated_at: string
}

export interface InsufficientStockItem {
  product: Product
  quantity: number
  inventory: number
}

export interface CreateOrderResult {
  success: true
  data: CreateOrderResponse
}

export interface CreateOrderConflict {
  success: false
  items: InsufficientStockItem[]
}

export interface CreateOrderResponse {
  client_secret: string
  order_id: string
}
